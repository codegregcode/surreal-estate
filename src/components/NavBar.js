import React from "react";
import "../styles/navbar.css";
import logo from "../assets/logo.png";

const NavBar = () => {
  return (
    <div className="navbar">
      <img src={logo} alt="logo" />
      <ul className="navbar-links">
        <li className="navbar-links-item">View Properties</li>
        <li className="navbar-links-item">Add a Property</li>
      </ul>
    </div>
  );
};

export default NavBar;
