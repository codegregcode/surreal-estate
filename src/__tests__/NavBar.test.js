import React from "react";
import { render, screen } from "@testing-library/react";
import NavBar from "../components/NavBar";

describe("NavBar", () => {
  const { asFragment } = render(<NavBar />);

  it("renders navbar component", () => {
    expect(asFragment()).toMatchSnapshot();
  });

  it("contains list/link items", () => {
    render(<NavBar />);
    const viewProperties = screen.getByText(/View Properties/);
    const addAProperty = screen.getByText(/Add a Property/);

    expect(viewProperties).toBeInTheDocument();
    expect(addAProperty).toBeInTheDocument();
  });

  it("renders logo", () => {
    render(<NavBar />);
    const logoAlt = screen.getByAltText(/logo/);

    expect(logoAlt).toBeInTheDocument();
  });
});
