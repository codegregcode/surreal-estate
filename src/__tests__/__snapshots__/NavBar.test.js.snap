// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders navbar component 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <img
      alt="logo"
      src="logo.png"
    />
    <ul
      class="navbar-links"
    >
      <li
        class="navbar-links-item"
      >
        View Properties
      </li>
      <li
        class="navbar-links-item"
      >
        Add a Property
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
